### PR TITLE
feat(location)!: `latitude`/`longitude` returns number

### PR DESCRIPTION
**Pull request by xDivisionByZerox**
This PR changes to return type of `address.latitude()` and `address.longitude()` from `string` to `number`.

Related to #1058.

This is a breaking change! Do not merge until the next major release.

### CODE REVIEW

**Review by ST-DDT — DISMISSED** [review]

File: src/modules/address/index.ts [hunk 1]
```diff
@@ -326,17 +326,16 @@ export class Address {
...
-        max,
-        precision: parseFloat(`${(0.0).toPrecision(precision)}1`),
-      })
-      .toFixed(precision);
+  // TODO @xDivisionBYZerox 2022-06-12 this signature should probably be an object for easyer maintainability
+  latitude(max: number = 90, min: number = -90, precision: number = 4): number {
+    return this.faker.datatype.number({
+      min,
+      max,
+      precision: parseFloat(`${(0.0).toPrecision(precision)}1`),
```
Comment (Shinigami92) [inline src/modules/address/index.ts:337]: Oof, I never realised this line, but IMO it's somewhat overengineered 🤔
Maybe we should just use the precision that is supported by JS? I think it's ~14 places?!
But on the other side, I don't totally care 🤷 I will leave this decision to other maintainers.

Comment (xDivisionByZerox) [reply src/modules/address/index.ts:337]: I mean, I get that you want to discuss that, but I don't think this is related to this PR and should be discussed in a separate discussion/issue.

Comment (Shinigami92) [reply src/modules/address/index.ts:337]: Ah yeah, I think you mean, because it was as implemented before. You may be right about that.

**Comment by ST-DDT** [comment]
Considering the other PR, we should consider whether we want to introduce a format parameter instead.

File: src/modules/address/index.ts [hunk 2]
```diff
@@ -326,17 +326,16 @@ export class Address {
...
    */
-  latitude(max: number = 90, min: number = -90, precision: number = 4): string {
-    return this.faker.datatype
-      .number({
-        min,
-        max,
-        precision: parseFloat(`${(0.0).toPrecision(precision)}1`),
-      })
-      .toFixed(precision);
+  // TODO @xDivisionBYZerox 2022-06-12 this signature should probably be an object for easyer maintainability
```
[the remark this answers is not in the document]
Comment (xDivisionByZerox) [reply src/modules/address/index.ts:332]: Nope, I'm just stupid (:

**Comment by xDivisionByZerox** [comment]
> Considering the other PR, we should consider whether we want to introduce a format parameter instead.

I really dislike the idea. For me, it doesn't make sense that coordinates would be strings. But we (as a library) should allow stringified strings as input parameters due to the nature of JS and enhancing the dx when using Faker.

File: src/modules/address/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (xDivisionByZerox) [reply src/modules/address/index.ts]: No, not intentional, thanks.

**Comment by ST-DDT** [comment]
Cross referencing related PRs: 
- https://github.com/faker-js/faker/pull/1066
- https://github.com/faker-js/faker/pull/1064
- https://github.com/faker-js/faker/pull/1061

**Comment by ST-DDT** [comment]
> I really dislike the idea. For me, it doesn't make sense that coordinates would be strings. But we (as a library) should allow stringified strings as input parameters due to the nature of JS and enhancing the dx when using Faker.

Further discussion regarding that topic please in https://github.com/faker-js/faker/pull/1066 or a separate issue/discussion.

**Review by ST-DDT — DISMISSED** [review]

**Review by ST-DDT — DISMISSED** [review]

**Comment by Shinigami92** [comment]
I eagerly put the PR title to scope `location` as we will solve 
- #1344

in v8.0 and this will then be grouped together

File: src/modules/address/index.ts [hunk 3]
```diff
@@ -384,19 +384,18 @@ export class AddressModule {
...
    */
-  latitude(max: number = 90, min: number = -90, precision: number = 4): string {
-    return this.faker.datatype
-      .number({
-        min,
-        max,
-        precision: parseFloat(`${(0.0).toPrecision(precision)}1`),
-      })
-      .toFixed(precision);
+  // TODO @xDivisionByZerox 2022-06-12 this signature should probably be an object for easier maintainability
```
Comment (Shinigami92) [inline src/modules/address/index.ts:268]: Should be solved via
- #1349

**Review by Shinigami92 — DISMISSED** [review]

**Review by ST-DDT — DISMISSED** [review]

**Review by ejcheng — DISMISSED** [review]

**Comment by Shinigami92** [comment]
@xDivisionByZerox needs rebase

**Comment by Shinigami92** [comment]
I tried to solve it, via the GitHub UI
Will fix it...

**Review by Shinigami92 — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**